x/evm/keeper/utils.go

```diff
@@ -11,19 +11,44 @@ import (
 
 	evmtypes "github.com/evmos/ethermint/x/evm/types"
 
+	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core"
 	ethtypes "github.com/ethereum/go-ethereum/core/types"
 )
 
-// DeductTxCostsFromUserBalance it calculates the tx costs and deducts the fees
-func (k Keeper) DeductTxCostsFromUserBalance(
+// DeductTxCostsFromUserBalance deducts the fees from the user balance. Returns an
+// error if the specified sender address does not exist or the account balance is not sufficient.
+//
+// TODO: Currently, Goland raises the warning that there are methods defined on both *keeper and keeper, which is not recommended: https://go.dev/doc/faq#methods_on_values_or_pointers
+func (k *Keeper) DeductTxCostsFromUserBalance(
```

Comment:
This func should be in `keeper.go`, not `utils.go`. Personally, I would also move `VerifyFee` and `CheckSenderBalance` to become Keeper methods—I think it's better abstraction.

Reply:
I agree that it makes for better abstraction if we turn those two functions into methods as well, posed this as a question in my previous review too - @danburck @fedekunze would you agree? 

Re: moving the methods, this would be my proposition put them:
- `func (k *Keeper) DeductTxCostsFromUserBalance` -> `x/evm/keeper/fees.go`
- `func (Keeper) VerifyFee` (turned into method) -> `x/evm/keeper/fees.go`
- `func (Keeper) CheckSenderBalance` (turned into method) -> stays in `utils.go`

Reply:
sounds good

Reply:
Are you sure we should move `VerifyFee` to the types module? Because we only ever use `VerifyFee` in conjunction with the Keeper methods so I would prefer moving the two fee related functions to a file `fees.go` inside of the keeper package and then keep `CheckSenderBalance` in the utils file?